### DISCUSSION TITLE
Regarding Conda and pip package distribution

### DISCUSSION THREAD

**Discussion by Raahul-Singh** [pnnl/neuromancer#88] [General]
Hi!
I'd like to help in the pip packaging of neuromancer.
Has any work started on this already?
If not, is there any plan to do so?

Thank you so much for this project!

**Comment by drgona** [comment]
Hi @Raahul-Singh, thank you for your interest in the library.

We currently support pip install from local toml file.
https://github.com/pnnl/neuromancer#pip-install

Packaging and hosting the library on PyPI is on the development plan, but currently, we don't have the resources allocated to this task. Therefore any proposed solution would be highly appreciated.

**Comment by Raahul-Singh** [comment]
In that case, let's start with a simple [twine](https://twine.readthedocs.io/en/latest/) based pypi upload.

First, we'd need to set up an account on [PyPI](https://pypi.org/account/register/) and optionally on [Test PyPI](https://test.pypi.org/account/register/) (useful for testing package uploads)
For building the wheels, we can use [build](https://pypa-build.readthedocs.io/en/latest/)
This will produce files in the `dist` directory.

I've, built and tested the distribution installation locally.

We can then upload using [twine](https://twine.readthedocs.io/en/latest/#using-twine)

For future updates, we'll to change the version number in pyproject.toml.

I feel this is the simplest way to start with the distribution. Should we proceed with this?

**Comment by madelynshapiro** [comment]
[NeuroMANCER is now available on PyPi!](https://pypi.org/project/neuromancer/)
